src/qt/transactiontablemodel.cpp

```diff
@@ -167,10 +167,28 @@ class TransactionTablePriv
         case CT_UPDATED:
             // Miscellaneous updates -- nothing to do, status update will take care of this, and is only computed for
             // visible transactions.
+            for (int i = lowerIndex; i < upperIndex; i++) {
+                TransactionRecord *rec = &cachedWallet[i];
+                rec->status.needsUpdate = true;
+            }
+            Q_EMIT parent->dataChanged(parent->index(lowerIndex, TransactionTableModel::Status), parent->index(upperIndex, TransactionTableModel::Status));
             break;
         }
     }
 
+    void updateAddressBook(const QString& address, const QString& label, bool isMine, const QString& purpose, int status)
+    {
+        std::string address2 = address.toStdString();
+        int index = 0;
+        for (auto& rec : cachedWallet) {
+            if (rec.strAddress == address2) {
+                rec.status.needsUpdate = true;
+                Q_EMIT parent->dataChanged(parent->index(index, TransactionTableModel::Status), parent->index(index, TransactionTableModel::Status));
```

Comment:
Btw.. Shouldn't this be
```suggestion
                Q_EMIT parent->dataChanged(parent->index(index, TransactionTableModel::ToAddress), parent->index(index, TransactionTableModel::ToAddress));
```
?

Reply:
Oh, yepp this is true. Fixing it now